lib/puppet/util/windows/taskscheduler2.rb

```diff
@@ -0,0 +1,310 @@
+require 'puppet/util/windows'
+
+# The TaskScheduler2 class encapsulates taskscheduler settings and behavior using the v2 API
+# https://msdn.microsoft.com/en-us/library/windows/desktop/aa383600(v=vs.85).aspx
+
+# @api private
+module Puppet::Util::Windows::TaskScheduler2
```

Comment:
Need to split this into component module statements as per @Iristyle recommendation

Reply:
I thought @DLuCJ ran into requiring things to be split, but the PR that got merged didn't have the changes I was referring to (https://github.com/puppetlabs/puppet/pull/5114/files and https://github.com/puppetlabs/puppet/pull/5098/files) - so I seem to be misremembering there.

The reason to do this is basic on Ruby scoping rules - see https://github.com/bbatsov/ruby-style-guide#namespace-definition for the surprising behavior!